src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -263,7 +263,7 @@ public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool
             HttpClient httpClient = CreateHttpClient();
 
             // NuGet packages are case insensitive
-            string endpoint = $"{NUGET_DEFAULT_CONTENT_ENDPOINT.EnsureTrailingSlash()}{purl.Name.ToLowerInvariant()}/{purl.Version.ToLowerInvariant()}/{purl.Name.ToLowerInvariant()}.nuspec";
+            string endpoint = $"{NUGET_DEFAULT_CONTENT_ENDPOINT.EnsureTrailingSlash()}{purl.Name.ToLowerInvariant()}/{nugetVersion}/{purl.Name.ToLowerInvariant()}.nuspec";
```

Comment:
Note that we have `this.Actions.DoesPackageExistAsync()` available to us here, which would allow us to defer to the NuGet SDK instead of maintaining this HTTP lookup ourselves. Would we be able to re-write this method to use that instead?

Reply:
Looks much better, thanks!